### PR TITLE
Add bert-base-uncased-conll2003_pfeiffer (#8)

### PR DESCRIPTION
**Pull request by vblagoje**
Adds a phrase chunker trained on conll2003 chunk tags. See https://github.com/davidsbatista/NER-datasets/tree/master/CONLL2003/ for more details. The trained adapter model aptly named "chunker" has F1 accuracy of ~91 on eval dataset and ~89 on the test dataset. Example notebook for noun phrase extraction is at https://bit.ly/33Nh84t

### CODE REVIEW

**Comment by arueckle** [comment]
Thanks for the PR and nice colab notebook!

Could you change the `default_version: "AdapterFusion"`  to something like `default_version: "1"` (also the file version). You might also want to remove the bibtex (as the adapter is not related to the listed paper).

**Comment by vblagoje** [comment]
> Could you change the `default_version: "AdapterFusion"` to something like `default_version: "1"` (also the file version). You might also want to remove the bibtex (as the adapter is not related to the listed paper).

Done @arueckle . I did not add any new commits, forced push this minor change.

**Comment by stefan-it** [comment]
Really interesting example :) Do you have a performance comparison between adapters and fine-tuning for this chunking task?

I did some extensive training for NER, but the performance difference is ~1% (CoNLL and WNUT) + varied learning rate and number of epochs, but I could never reach the fine-tuning result.

**Comment by vblagoje** [comment]
@stefan-it Tbh I didn't spend a lot of time tuning the model. I am still learning the subtle intricacies. The full HF model had F1 score in the range of ~93. All the code is at https://github.com/vblagoje/adapter-transformers/tree/token_classification Please feel free to have a look. I did a minor [refactoring](https://github.com/vblagoje/adapter-transformers/commit/624c4ad461faf1834a4ddf50b8b0cde7d7c99b8f) of token classification so I can easily add any type of token classification task. I'll see if HF people want this change, I think it makes more sense to have it there than to diverge token classification in adapter fork. LMK your thoughts.

File: tasks/text_task/chunker.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+task: chunker
```
Comment (calpt) [inline tasks/text_task/chunker.yaml]: Maybe a bit petty but I'd prefer `chunking` instead of `chunker` to be more consistent with other task names. :)

**Review by calpt — COMMENTED** [review]
@vblagoje  PR looks good to me. Really like your colab notebooks too!

File: tasks/text_task/chunker.yaml (hunk 1 above)
Comment (vblagoje) [reply tasks/text_task/chunker.yaml]: Thanks, @calpt and no problem, I'll fix it tomorrow! Let's hold the merge until then.

Comment (vblagoje) [reply tasks/text_task/chunker.yaml]: @calpt now that I am looking closer at all the task names both in terms of YAML file naming and display names, isn't a noun `chunk` a better match for YAML file name rather than a verb `chunking`, just like `qa`, `sentiment`, `commonsense` etc. However,  just like Question Answering, Sentiment Analysis, etc I'll name the task (display name) Phrase Chunking. LMK

Comment (vblagoje) [reply tasks/text_task/chunker.yaml]: I wanted to try out the notebook with a new zip file `chunk-conll2003.zip` and task renamed to `chunk`. I thought I only needed to change the name of the task in the zip file (adapter_config.json) and adjust the name of the model adapter passed to model inference call. Although inference call worked I got a response that was totally off. Interested what is going on @calpt ?!

Comment (vblagoje) [reply tasks/text_task/chunker.yaml]: I have a preview of changes on another [branch](https://github.com/vblagoje/Hub/commit/870b596ff8fd2edbd4b903ace3cb8a66aa754754) The notebook now works. I just retrained the model with a new adapter name. LMK @calpt

Comment (calpt) [reply tasks/text_task/chunker.yaml]: > isn't a noun chunk a better match for YAML file name rather than a verb chunking

Yes, that's also fine. While your're at it, maybe quickly also change `POS Tagger` to `POS Tagging` (sorry, didn't pay attention to this in your last PR). Thanks!

>  I thought I only needed to change the name of the task in the zip file (adapter_config.json) and adjust the name of the model adapter passed to model inference call

The name in the json file is used to identify the correct weights when loading and to insert them in the correct places in the model. So, when changing the name in the json, the name must also be changed in the pickled weights file. Something like this should do the trick (using `load_as`):

```python
model.load_adapter("/load/path", load_as="chunk")
model.save_adapter("/save/path", "chunk")
```

and as sidenote: the name in the json doesn't have to match the task name in the yaml, so just renaming the task in the yaml file would've been enough.

> I have a preview of changes on another branch The notebook now works. I just retrained the model with a new adapter name.

Looks good, just add it to this PR.

Comment (vblagoje) [reply tasks/text_task/chunker.yaml]: Thanks for the weights tips @calpt . If we are changing `POS Tagger` to `POS Tagging` I used the opportunity to set the default_version to 1 in `adapters/vblagoje/bert-base-uncased-ldc2012t13_pfeiffer.yaml`. Please look at that change as well. Next time I'll just add commits to PR rather than force push. LMK if I should adjust anything else.

Comment (calpt) [reply tasks/text_task/chunker.yaml]: > Next time I'll just add commits to PR rather than force push.

I'll squash them anyway during merging, so this really doesn't make a difference in the end :)

As we somehow got back to the POS tagging adapter already, the very last little step to perfection would be to also remove the citation there (as @arueckle mentioned above for this adapter). That's definitely the last thing from my side, rest looks very good!

Comment (vblagoje) [reply tasks/text_task/chunker.yaml]: No problem @calpt Done

Comment (calpt) [reply tasks/text_task/chunker.yaml]: @vblagoje Thanks! Merging now.

**Comment by vblagoje** [comment]
> @vblagoje PR looks good to me. Really like your colab notebooks too!

Thanks @calpt . It was a pleasure to experiment and work on these tasks!
